### PR TITLE
Dok Matrix finito (?) 

### PR DESCRIPTION
**Pull request by zdrever**
Everything except .power is working for DoK (Everything in scope that is) 

#47 and #27 are resolved

opened #59

### CODE REVIEW

File: test/dok_matrix_test.rb [hunk 1]
```diff
@@ -146,10 +146,50 @@ def test_multiply
...
+    exp = @matrix * test
+    assert_equal(exp, actual.to_matrix, 'Matrix multiplication failed')
+  end
+
   def test_power
-    scalar = rand(-10..10)
-    actual = @sparse_matrix**scalar
-    exp = @matrix**scalar
-    assert_equal(exp, actual.to_matrix, 'Matrix exponentiation failed')
+    scalar = rand(0..10)
```
Comment (nklapste) [inline test/dok_matrix_test.rb]: I would make two tests one for a regular and one for a irregular matrix to validate matrix power.
(These don't necessarily have to be based of randomly generated like this test).

**Review by nklapste — APPROVED** [review]

File: test/dok_matrix_test.rb (hunk 1 above)
Comment (figintern) [reply test/dok_matrix_test.rb]: Better not to use rand(0..10) anyways as generating 1 will be a trivial case and always pass.

Comment (figintern) [reply test/dok_matrix_test.rb]: I used rand (2..5) in my own PR for csr_matrix

File: lib/dok_matrix.rb [hunk 2]
```diff
@@ -52,13 +52,6 @@ def []=(row, col, value)
...
 
-  private def set_value(row, col, value)
-    raise(ArgumentError) unless row >= 0 && col >= 0
-    raise(TypeError) unless value.is_a?(Numeric) || value.nil?
-
-    @dict[[row, col]] = value
-  end
-
   def **(other)
     raise(TypeError) unless other.is_a?(Numeric)
```
Comment (figintern) [inline lib/dok_matrix.rb:57]: I believe the line below should read 'super other'? Why is it 'super**other'

Comment (zdrever) [reply lib/dok_matrix.rb:57]: Ya you correct. Fixing

File: test/dok_matrix_test.rb (hunk 1 above)
Comment (zdrever) [reply test/dok_matrix_test.rb]: I can fix these as well.. 

Is there a way to specify how many times test cases should be run?

**Pull request merged**